nodejs/dist/uws.js

```diff
@@ -1,98 +1,132 @@
+     */
+    broadcast(message, options) {
         /* only listen to binary option */
-        this.nativeServer.broadcast(message, options.binary);
-    };
-
-    /* original connection callback */
-    this.connectionCallback = function (nativeSocket) {
-        var socket = new Socket(nativeSocket, self);
-        self.nativeServer.setData(nativeSocket, socket);
-        self.emit('connection', socket);
-    };
-
-    /* this function needs to handle the upgrade directly! */
-    this.handleUpgrade = function (request, socket, upgradeHead, callback) {
+        this.nativeServer.broadcast(message, options && options.binary || false);
+    }
+
+    /**
+     * Handles a HTTP Upgrade request.
+     *
+     * @param {http.IncomingMessage} request HTTP request
+     * @param {net.Socket} Socket between server and client
+     * @param {Buffer} upgradeHead The first packet of the upgraded stream
+     * @param {Function} callback Callback function
+     * @public
+     */
+    handleUpgrade(request, socket, upgradeHead, callback) {
         /* register a special connection handler */
-        self.nativeServer.onConnection(function (nativeSocket) {
-            var socket = new Socket(nativeSocket, self);
-            self.nativeServer.setData(nativeSocket, socket);
+        this.nativeServer.onConnection((nativeSocket) => {
+            const sock = new Socket(nativeSocket, this);
+            this.nativeServer.setData(nativeSocket, sock);
 
             /* internal variables */
-            socket.upgradeReq = request;
-            socket.readyState = socket.OPEN;
-            callback(socket);
+            sock.upgradeReq = request;
+            callback(sock);
 
             /* todo: reset connection handler when the upgrade queue is empty */
             /* this will probably never be noticed as you don't mix upgrade handling */
         });
 
         /* upgrades will be handled immediately */
-        self.nativeServer.upgrade(socket._handle.fd, request.headers['sec-websocket-key']);
+        this.nativeServer.upgrade(socket._handle.fd, request.headers['sec-websocket-key']);
         socket.destroy();
-    };
-
-    this.nativeServer.onConnection(this.connectionCallback);
-    this.nativeServer.onDisconnection(function (nativeSocket) {
-        var socket = self.nativeServer.getData(nativeSocket);
-        socket.emit('close');
-        /* make sure to clear any set data */
-        self.nativeServer.setData(nativeSocket);
-    });
-
-    this.nativeServer.onMessage(function (nativeSocket, message, binary) {
-        var socket = self.nativeServer.getData(nativeSocket);
-        if (binary) {
-            socket.emit('message', message);
-        } else {
-            /* we are supposed to pass text as strings */
-            socket.emit('message', message.toString());
-        }
-    });
-};
-
-util.inherits(Socket, EventEmitter);
-util.inherits(module.exports.Server, EventEmitter);
+    }
+}
+
+exports.Server = Server;
+exports.uws = uws;
```

Comment:
I'm exposing the original interface here for people that don't want to use the `ws` interface.
This might be useful for embedders like Primus where the `Spark` is already wrapping the `nativeSocket`.